readthedocs/projects/admin.py

```diff
@@ -88,6 +119,46 @@ def send_owner_email(self, request, queryset):
 
     send_owner_email.short_description = 'Notify project owners'
 
+    def ban_owner(self, request, queryset):
+        """Ban project owner
+
+        This will only ban single owners, because a malicious user could add a
+        user as a co-owner of the project. We don't want to induce and
+        collatoral damage when flagging users.
+        """
+        total = 0
+        for project in queryset:
+            if project.users.count() == 1:
+                count = (UserProfile.objects
+                         .filter(user__projects=project)
+                         .update(banned=True))
+                total += count
+            else:
+                messages.add_message(request, messages.ERROR,
+                                     'Project has multiple owners: {0}'.format(project))
+        if total == 0:
+            messages.add_message(request, messages.ERROR, 'No users banned')
+        else:
+            messages.add_message(request, messages.INFO,
+                                 'Banned {0} user(s)'.format(total))
+
+    ban_owner.short_description = 'Ban project owner'
+
+    def delete_selected_and_artifacts(self, request, queryset):
+        if request.POST.get('post'):
+            for project in queryset:
+                broadcast(type='app', task=remove_dir, args=[project.doc_path])
+        return delete_selected(self, request, queryset)
```

Comment:
Would be good to add a docstring about why we're doing this. To clean up left over stuff from build servers? We could also just call .delete() directly, which should trigger the logic that does this, instead of special casing this one cleanup task.

Reply:
The admin calls `.delete()` on the `Project` queryset, bulking the operations, this was to avoid messing with any of the logic there.

`Project` also doesn't have a delete method that does this currently, because we need to make a decision on whether a user deleting a project should remove the files from webs.